misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobQueue.kt

```
@@ -97,6 +103,102 @@ internal class SqsJobQueue @Inject internal constructor(
+                  key,
+                  MessageAttributeValue()
+                    .withDataType("String")
+                    .withStringValue(value)
+                )
+              }
+
+              val metadata = mutableMapOf(
+                SqsJob.JOBQUEUE_METADATA_ORIGIN_QUEUE to queueName.parentQueue.value,
+                SqsJob.JOBQUEUE_METADATA_IDEMPOTENCE_KEY to message.idempotenceKey
+              )
+
+              // Preserve original trace id, if available.
+              (span as? DDSpan)?.let {
+                val traceId = it.context().traceId.toString()
+                metadata[SqsJob.JOBQUEUE_METADATA_ORIGINAL_TRACE_ID] = traceId
+                // TODO(bruno): drop this attribute after rollout; moved to metadata
+                addMessageAttributesEntry(
+                  SqsJob.ORIGINAL_TRACE_ID_ATTR,
+                  MessageAttributeValue().withDataType("String").withStringValue(traceId)
+                )
+              }
+
+              addMessageAttributesEntry(
+                SqsJob.JOBQUEUE_METADATA_ATTR,
+                MessageAttributeValue()
+                  .withDataType("String")
+                  .withStringValue(moshi.adapter<Map<String, String>>().toJson(metadata))
+              )
+            }
+          }
+
+          timed {
+            client.sendMessageBatch(SendMessageBatchRequest(queue.url, messageEntries))
+          }
+        }
+
+        metrics.sqsSendTime.record(
+          sendDuration.toMillis().toDouble(),
+          queueName.value,
+          queueName.value
+        )
+
+        val batchResult = JobQueue.BatchEnqueueResult(result.successful.map { it.id }, result.failed.map {
+          JobQueue.EnqueueErrorResult(
+            it.id,
+            it.isSenderFault,
+            it.code,
+            it.message
+          )
+        })
+
+        if(batchResult.failed.isNotEmpty()){
+          log.error { "failed to batch enqueue to ${queueName.value}" }
```

is this the right place to log an error? If it’s truly an error should this function throw instead of returning normally?

---

fair point, I was following the pattern of the single enqueue. In this case since the client gets the successful & failed messages they can choose to log